### PR TITLE
Add support for Buildkite environment

### PR DESCRIPTION
**Pull request by lox**
We support parallelism on Buildkite that seems like it should work nicely with the groups support here. This adds support for the relevant env.

More details at https://buildkite.com/docs/tutorials/parallel-builds.

Ruby is not my native language, so please let me know if I'm doing it wrong.

### CODE REVIEW

File: lib/forking_test_runner.rb [hunk 1]
```diff
@@ -176,8 +176,16 @@ def record_test_runtime(mode, results, log)
...
-        id = ENV.fetch("TRAVIS_BUILD_NUMBER")
+        if ENV.has_key? "TRAVIS_REPO_SLUG" &&  ENV.has_key? "TRAVIS_BUILD_NUMBER"
+          slug = ENV.fetch("TRAVIS_REPO_SLUG").sub("/", "-")
+          id = ENV.fetch("TRAVIS_BUILD_NUMBER"
+        elsif ENV.has_key? "BUILDKITE_PIPELINE_SLUG" && ENV.has_key? "BUILDKITE_JOB_ID"
+          slug = ENV.fetch("BUILDKITE_ORG_SLUG") + "-" + ENV.fetch("BUILDKITE_PIPELINE_SLUG")
+          id = ENV.fetch("BUILDKITE_JOB_ID")
+        else
+          raise "No env found for a CI provider"
+        done
```
Comment (grosser) [inline lib/forking_test_runner.rb]:
```suggestion
        end
```

**Comment by grosser** [comment]
just a tiny typo :D

... I'm not really a fan of adding all kinds of provider specific code, but otherwise we have to add new options .. so 🤷‍♂️

**Comment by grosser** [comment]
... also CI seems to not be running :/

**Comment by grosser** [comment]
... travis says "Abuse detected" :D

**Comment by lox** [comment]
At some point might be worth extracting provider specific options into something similar to https://buildkite.com/smale/samson-build/builds/60, but this seemed like the smallest possible incremental fix.

**Comment by grosser** [comment]
travis said they fixed it 30 min ago, so I'll just take this an fix it up on master

**Comment by grosser** [comment]
fixed a few more typos and added missing tests v1.4.0

**Comment by lox** [comment]
Appreciate it @grosser, I was going to follow up with some tests! ❤️

**Pull request merged**